### PR TITLE
Add javadoc to ReactiveMessageSenderBuilder

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: pulsar-client-reactive-api/src/main/java/org/apache/pulsar/reactive/client/api/ReactiveMessageSenderBuilder.java [hunk 1]
```diff
@@ -28,156 +28,555 @@
...
+ * Builder interface for {@link ReactiveMessageSender}.
+ *
+ * @param <T> the message payload type
+ * @author Lari Hotari
+ * @author Christophe Bornet
+ */
 public interface ReactiveMessageSenderBuilder<T> {
 
+	/**
+	 * Sets the cache to use for the sender.
```
Comment (lhotari) [inline pulsar-client-reactive-api/src/main/java/org/apache/pulsar/reactive/client/api/ReactiveMessageSenderBuilder.java:43]: The ReactiveMessageSenderCache will need a better explanation, but we could do that in a separate PR.

File: pulsar-client-reactive-api/src/main/java/org/apache/pulsar/reactive/client/api/ReactiveMessageSenderBuilder.java [hunk 2]
```diff
@@ -28,156 +28,555 @@
...
+	 * @return the sender builder instance
+	 */
 	ReactiveMessageSenderBuilder<T> cache(ReactiveMessageSenderCache producerCache);
 
+	/**
+	 * Sets the maximum number of in-flight messages for the sender. When this value is
+	 * reached, backpressure will be triggered on the
+	 * {@link ReactiveMessageSender#sendOne(MessageSpec)}/{@link ReactiveMessageSender#sendMany(Publisher)}
+	 * operations. Note that the maxInflight setting applies globally for all the
+	 * operations called on the sender.
```
[the remark this answers is not in the document]
Comment (lhotari) [reply pulsar-client-reactive-api/src/main/java/org/apache/pulsar/reactive/client/api/ReactiveMessageSenderBuilder.java:54]: Added #38

**Review by lhotari — APPROVED** [review]

File: pulsar-client-reactive-api/src/main/java/org/apache/pulsar/reactive/client/api/ReactiveMessageSenderBuilder.java (hunk 1 above)
Comment (cbornet) [reply pulsar-client-reactive-api/src/main/java/org/apache/pulsar/reactive/client/api/ReactiveMessageSenderBuilder.java:43]: Yes I'm doing it interface by interface 🙂
